petals-esb-deployer-impl/src/main/java/org/ow2/petals/deployer/runtimemodel/RuntimeModel.java

```diff
@@ -53,8 +58,11 @@ public RuntimeContainer getContainer(final String id) {
         return containers.get(id);
     }
 
+    /**
+     * Adding or removing from the returned collection does not affect the model.
+     */
     public Collection<RuntimeContainer> getContainers() {
-        return containers.values();
+        return new HashSet<>(containers.values());
```

Comment:
I think you can find better using class Collections.unmodifiableCollection() to avoid adding and removing from the returned collection.